Feature/ #636 Module Specific Readme Files (#664)
WordPress/performance: modules/images/webp-uploads/readme.txt

=== WebP Uploads ===

Contributors:      wordpressdotorg
Requires at least: 6.1
Tested up to:      6.1
Requires PHP:      5.6
Stable tag:        1.0.0
License:           GPLv2 or later
License URI:       https://www.gnu.org/licenses/gpl-2.0.html
Tags:              performance, images, webp

Creates WebP versions for new JPEG image uploads if supported by the server.

== Description ==

> This plugin adds WebP support for media uploads within the WordPress application. WebP images will be generated only for new uploads, preexisting imagery will not be converted to WebP format. By default, WebP images will only be generated for JPEG uploads, only the original uploaded file will still exist as a JPEG image. ALl generated image sizes will exist as WebP only. If you wish to change this behaviour, there is a checkbox in `Settings > Media` that - when checked - will alter the behaviour of this plugin to generate both JPEG and WebP images for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making said change).

Reviewer: ```suggestion
This plugin adds WebP support for media uploads within the WordPress application. WebP images will be generated only for new uploads, pre-existing imagery will not be converted to WebP format. By default, WebP images will only be generated for JPEG uploads, only the original uploaded file will still exist as a JPEG image. All generated image sizes will exist as WebP only. If you wish to change this behaviour, there is a checkbox in `Settings > Media` that - when checked - will alter the behaviour of this plugin to generate both JPEG and WebP images for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making said change).
```

Author: Thanks @mukeshpanchal27 , updated

=== WebP Uploads ===

Contributors:      wordpressdotorg
Requires at least: 6.1
Tested up to:      6.1
Requires PHP:      5.6
Stable tag:        1.0.0
License:           GPLv2 or later
License URI:       https://www.gnu.org/licenses/gpl-2.0.html
Tags:              performance, images, webp

Creates WebP versions for new JPEG image uploads if supported by the server.

== Description ==

> This plugin adds WebP support for media uploads within the WordPress application.

Reviewer 2: Technically not a blocker, but since this is pretty much the only module / plugin that has a UI to it, we still may want to provide at least a bit more explanation in 2-3 short paragraphs. Things to potentially highlight:
* WebP images will be generated only for new uploads
* By default, only WebP images will be generated for JPEG uploads, only the original uploaded file will still exist as JPEG
* There is a checkbox in `Settings > Media` that can change the behavior though to generate _both_ JPEG and WebP for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making that change)

Author: @felixarntz (cc @mukeshpanchal27 ) this is ready for re-review following changes pushed in recent commit(s).

= I've activated the WebP Uploads module, but WebP images are not always generated when I upload a JPEG image. Why? =

There are two primary reasons that a WebP image may not be generated:

1. The WebP Uploads plugin has identified that the WebP version of the uploaded JPEG image would have a larger file size than the original JPEG image, so it does not generate the WebP version.
2. The JPEG image was not uploaded to the [Media Library](https://wordpress.com/support/media/). At this time, WebP versions are only generated for images to the Media Library. WebP versions are not generated for JPEG images that are added to your site in other ways, such as in a template file or the [Customizer](https://wordpress.com/support/customizer/).

= With the WebP Uploads plugin activated, will the plugin generate JPEG and WebP versions of every image that I upload? =

By default, the WebP Uploads plugin will only generate WebP versions of the images that you upload. If you wish to have both WebP **and** JPEG versions generated, you can navigate to **Settings > Media** and enable the **Generate JPEG files in addition to WebP** option.

== Changelog ==

> = 1.0.0 =
> 
> **Initial Release**

Reviewer 2: No need for an extra "heading" here
```suggestion
= 1.0.0 =
```

=== WebP Uploads ===

Contributors:      wordpressdotorg
Requires at least: 6.1
Tested up to:      6.1
Requires PHP:      5.6
Stable tag:        1.0.0
License:           GPLv2 or later
License URI:       https://www.gnu.org/licenses/gpl-2.0.html
Tags:              performance, images, webp

Creates WebP versions for new JPEG image uploads if supported by the server.

== Description ==

This plugin adds WebP support for media uploads within the WordPress application.

> 

Reviewer: ```suggestion
```
Remove extra line.

> === Performance Lab - WebP Uploads Module ===

Reviewer 2: The plugin should be called like the module, "WebP Uploads". It does not require Performance Lab, so shouldn't look as if it was an extension.

Contributors:      wordpressdotorg
Requires at least: 6.1
Tested up to:      6.1
Requires PHP:      5.6
Stable tag:        1.0.0
License:           GPLv2 or later
License URI:       https://www.gnu.org/licenses/gpl-2.0.html

> Tags:              performance, images, site health, measurement, webp

Reviewer: ```suggestion
Tags:              performance, images, webp
```

I don't think we have to add `site health` and `measurement` for WebP Uploads

> WebP uploads module for the [performance plugin](https://wordpress.org/plugins/performance-lab/) from the WordPress Performance Team.

Reviewer 2: I don't think we should mention the Performance Lab plugin here given that this is the description for the standalone plugin. I think we can use the description that is also present in the module header:
```suggestion
Creates WebP versions for new JPEG image uploads if supported by the server.
```

== Description ==

This plugin adds a WebP uploads module to the [performance plugin](https://wordpress.org/plugins/performance-lab/), enabling WebP support for media uploads within the WordPress application, when this module is enabled.

> This plugin is intended to be used alongside the [performance plugin](https://wordpress.org/plugins/performance-lab/), and will not function as a standalone plugin. The performance plugin may be obtained [here](https://wordpress.org/plugins/performance-lab/).

Reviewer 2: The latter is not correct, this plugin _will_ function as a standalone plugin. That's part of why we're breaking them apart.

I would avoid the mention of the Performance Lab plugin here and focus purely on a brief description of what this plugin does.

== Installation ==

> = Installation from within WordPress =
> 
> 1. Visit **Plugins > Add New**.
> 2. Search for **Performance Lab - WebP Uploads Module**.
> 3. Install and activate the **Performance Lab - WebP Uploads Module** plugin.
> 
> = Manual installation =
> 
> 1. Upload the entire `performance-lab-webp-support-module` folder to the `/wp-content/plugins/` directory.
> 2. Visit **Plugins**.
> 3. Activate the **Performance Lab - WebP Uploads Module** plugin.

Reviewer 2: See above:
```suggestion
= Installation from within WordPress =

1. Visit **Plugins > Add New**.
2. Search for **WebP Uploads**.
3. Install and activate the **WebP Uploads** plugin.

= Manual installation =

1. Upload the entire `webp-uploads` folder to the `/wp-content/plugins/` directory.
2. Visit **Plugins**.
3. Activate the **WebP Uploads** plugin.
```

= After activation =

1. Visit the new **Settings > Performance** menu.

> 2. Enable the individual modules you would like to use, one of which should be this WebP support module.

Reviewer 2: This should be removed, as the plugin will work as soon as it's activated.

== Frequently Asked Questions ==

= What is the purpose of this plugin? =

The primary purpose of the Performance Lab - WebP Support Module plugin is to add an additional module to the performance plugin that enables the ability to toggle on/off WebP support.

> 

Reviewer 2: I think we can remove this.

= Can I use this plugin on my production site? =

> Per the primary purpose of the plugin (see above), it can be considered stable for production usage given that this particular module is not marked as experimental.

Reviewer 2: This refers to a module, which it shouldn't. The standalone plugin will have no notion of what is experimental or not.

= Where can I submit my plugin feedback? =

> Feedback is encouraged and much appreciated, especially since this plugin may contain future WordPress core features. If you have suggestions or requests for new features, you can [submit them as an issue in the Performance Lab GitHub repository](https://github.com/WordPress/performance/issues/new/choose). If you need help with troubleshooting or have a question about the plugin, please [create a new topic on our support forum](https://wordpress.org/support/plugin/performance-lab/#new-topic-0).

Reviewer 2: ```suggestion
Feedback is encouraged and much appreciated, especially since this plugin may contain future WordPress core features. If you have suggestions or requests for new features, you can [submit them as an issue in the WordPress Performance Team's GitHub repository](https://github.com/WordPress/performance/issues/new/choose). If you need help with troubleshooting or have a question about the plugin, please [create a new topic on our support forum](https://wordpress.org/support/plugin/webp-uploads/#new-topic-0).
```

= How can I contribute to the plugin? =

Contributions are always welcome! Learn more about how to get involved in the [Core Performance Team Handbook](https://make.wordpress.org/performance/handbook/get-involved/).

> 

Reviewer 2: Can you add (slightly modified versions of) the two WebP Uploads specific entries in our current Performance Lab `readme.txt` here? Those are relevant and should be part of this readme. The mentioning of activating the module in Performance Lab should be removed from them though, as just by activating this plugin on its own those two questions may be relevant.

== Changelog ==

= 1.0.0 =

**Initial Release**

> * Initial release of the WebP support performance plugin module as a standalone plugin. ([636](https://github.com/WordPress/performance/issues/636))

Reviewer 2: We typically refer to PRs in the changelog instead of issues.
```suggestion
* Initial release of the WebP Uploads plugin as a standalone plugin. ([664](https://github.com/WordPress/performance/pull/664))
```
